File: Python/devpossible_ton/devpossible_ton/models/ton_enum.py
"""
TonEnum - Enum model for TON
Copyright (c) 2024 DevPossible, LLC
"""

from typing import Union, List


class TonEnum:
    """Enum model for TON format."""

    def __init__(self, value: Union[str, List[str]]):
        if isinstance(value, str):
            self.values = [value]
            self.is_single_value = True
        else:
            self.values = value
            self.is_single_value = False

    def get_value(self) -> Union[str, List[str]]:
        """Get the value(s)."""
        return self.values[0] if self.is_single_value else self.values

    def get_values(self) -> List[str]:
        """Get all values as list."""
        return self.values

    def contains(self, value: str) -> bool:
        """Check if contains a value."""
        return value in self.values

    def is_set(self) -> bool:
        """Check if this is an enum set."""
        return not self.is_single_value

    def to_json(self) -> Union[str, List[str]]:
        """Convert to JSON-serializable value."""
        return self.get_value()

    def __str__(self) -> str:
        """String representation."""
        if self.is_single_value:
            return f"|{self.values[0]}|"
        return ''.join(f"|{v}" for v in self.values) + '|'

File: JavaScript/devpossible-ton/src/models/TonEnum.ts
/**
 * TonEnum - Enum model for TON
 * Copyright (c) 2024 DevPossible, LLC
 */

export class TonEnum {
  private values: string[];
  private isSingleValue: boolean;

  constructor(value: string | string[]) {
    if (typeof value === 'string') {
      this.values = [value];
      this.isSingleValue = true;
    } else {
      this.values = value;
      this.isSingleValue = false;
    }
  }

  public getValue(): string | string[] {
    return this.isSingleValue ? this.values[0] : this.values;
  }

  public getValues(): string[] {
    return this.values;
  }

  public contains(value: string): boolean {
    return this.values.includes(value);
  }

  public isSet(): boolean {
    return !this.isSingleValue;
  }

  public toJSON(): string | string[] {
    return this.getValue();
  }

  public toString(): string {
    if (this.isSingleValue) {
      return `|${this.values[0]}|`;
    }
    return this.values.map(v => `|${v}`).join('') + '|';
  }
}